examples/medplum-demo-bots/src/deduplication/merge-bot.ts

```diff
@@ -0,0 +1,151 @@
+
+/**
+ * Links two patient records indicating one replaces the other.
+ *
+ * @param src - The source patient record which is being replaced.
+ * @param target - The target patient record which will replace the source.
+ * @returns - Object containing updated source and target patient records with their links.
+ */
+export function linkPatientRecords(src: Patient, target: Patient): MergedPatients {
+  const targetLinks = target.link ?? [];
+  targetLinks.push({ other: createReference(src), type: 'replaces' });
+  const targetCopy = deepClone(target);
+
+  const srcLinks = src.link ?? [];
+  srcLinks.push({ other: createReference(target), type: 'replaced-by' });
+  const srcCopy = deepClone(src);
+  return { src: { ...srcCopy, link: srcLinks, active: false }, target: { ...targetCopy, link: targetLinks } };
+}
+
+/**
+ * Merges contact information (identifiers) of two patient records.
+ * Identifiers from the source are marked as 'old' in the merged record.
+ *
+ * @param src - The source patient record.
+ * @param target - The target patient record.
+ * @param fields - Optional additional fields to be merged.
+ * @returns - Object containing the original source and the merged target patient records.
+ */
+export function mergePatientRecords(src: Patient, target: Patient, fields?: Partial<Patient>): MergedPatients {
+  const srcIdentifiers = src.identifier ?? [];
+  const mergedIdentifiers = srcIdentifiers.map((identifier) => ({
+    ...identifier,
+    use: 'old' as Identifier['use'],
+  }));
+  const targetCopy = deepClone(target);
+  targetCopy.identifier = [...(targetCopy.identifier ?? []), ...mergedIdentifiers];
+  const targedMerged = { ...targetCopy, ...fields };
+  return { src: src, target: targedMerged };
+}
+
+/**
+ * Updates the subject of clinical resources from the source patient to the target patient.
+ *
+ * @param medplum - Instance of the MedplumClient for server interactions.
+ * @param src - The source patient record.
+ * @param target - The target patient record.
+ * @param clinicalResource - The type of the clinical resource (e.g., 'ServiceRequest').
+ */
+export async function updateClinicalReferences<T extends ResourceWithSubject>(
+  medplum: MedplumClient,
+  src: Patient,
+  target: Patient,
+  clinicalResource: T['resourceType']
+): Promise<void> {
+  // Search for clinical resources related to the source patient.
+  const reports = await medplum.searchResources(clinicalResource, { subject: src });
+  (reports as ResourceWithSubject[]).map(async (report) => {
+    // Update each found resource's subject to the target patient.
+    report.subject = createReference(target);
+    await medplum.updateResource(report);
+  });
+}
+
+async function addToDoNotMatchList(
+  medplum: MedplumClient,
+  patientList: Reference<Patient>,
+  patientAdded: Reference<Patient>
+): Promise<void> {
+  const lists = await medplum.searchResources('List', { subject: patientList, code: 'doNotMatch' });
```

Comment:
might be easier to just use a `searchOne` here for convenience. Hopefully, we haven't created 2 doNoMatch lists for a patient